consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -70,30 +70,71 @@ where
         Self { branches, tips }
     }
 
-    #[must_use = "this returns the result of the operation, without modifying the original"]
-    fn apply_header(&self, header: Id, parent: Id, slot: Slot) -> Result<Self, Error<Id>> {
+    /// Create a new [`Branches`] instance with the updated state.
+    /// This method adds a [`Branch`] to the new instance without running
+    /// validation.
+    ///
+    /// # Warning
+    ///
+    /// **This method bypasses safety checks** and can corrupt the state if used
+    /// incorrectly.
+    /// Only use for recovery, debugging, or other manipulations where the input
+    /// is known to be valid.
+    ///
+    /// # Arguments
+    ///
+    /// * `header` - The ID of the block to be added.
+    /// * `parent` - The ID of the parent block. Due to the nature of the method
+    ///   (`unchecked`), the existence of the parent block is not verified.
+    /// * `slot` - The slot of the block to be added.
+    /// * `chain_length`: The position of the block in the chain.
+    #[must_use = "Returns a new instance with the updated state, without modifying the original."]
+    fn apply_header_unchecked(
+        &self,
+        header: Id,
+        parent: Id,
+        slot: Slot,
+        chain_length: u64,
+    ) -> Self {
         let mut branches = self.branches.clone();
+
         let mut tips = self.tips.clone();
-        // if the parent was the head of a branch, remove it as it has been superseded
-        // by the new header
+        // When running this method by itself (not through `Self::apply_header`), the
+        // parent should not in the tips set. That being said, having this call
```

Comment:
should not **be**? I think you missed something there. It actually do not matter it is or not for this method. Should be check outside right?